source/lac/trilinos_vector.cc

```diff
@@ -344,6 +380,7 @@ namespace TrilinosWrappers
         {
           v.block(block).trilinos_vector().ExtractCopy (entries, 0);
           entries += v.block(block).local_size();
+          owned_elements.add_indices(v.block(block).locally_owned_elements());
```

Comment:
Does this work? I would think that each block will count locally owned elements from 0 so you have to shift them here?